### ISSUE TITLE
OpenAPI.yaml with recursion

### ISSUE THREAD

**Issue by KaiSuchomel** [quarkusio/quarkus#36677]
### Describe the bug

With the current Quarkus (3.5.0.CR1) a recursion in the openapi.yaml file will occur. 
When using Enum in a Collection (in my case a Set) the following openapi.yaml is generated:

```yaml
openapi: 3.0.3
info:
  title: quarkus-openapi API
  version: 1.0.0-SNAPSHOT
paths:
  /hello:
    get:
      tags:
      - Greeting Resource
      responses:
        "200":
          description: OK
          content:
            text/plain:
              schema:
                type: string
components:
  schemas:
    MyClass:
      description: MyClass Description
      type: object
      properties:
        myEnums:
          description: My Enum Set
          uniqueItems: true
          type: array
          items:
            $ref: '#/components/schemas/MyEnum'
    MyEnum:
      $ref: '#/components/schemas/MyEnum'
```
The Enum MyEnum has a selfreference.

### Expected behavior

The openapi.yaml should look like the following:

```yaml
openapi: 3.0.3
info:
  title: quarkus-openapi API
  version: 1.0.0-SNAPSHOT
paths:
  /hello:
    get:
      tags:
      - Greeting Resource
      responses:
        "200":
          description: OK
          content:
            text/plain:
              schema:
                type: string
components:
  schemas:
    MyClass:
      description: MyClass Description
      type: object
      properties:
        myEnums:
          description: My Enum Set
          uniqueItems: true
          type: array
          items:
            $ref: '#/components/schemas/MyEnum'
    MyEnum:
      description: A special Enum
      enum:
      - VALUE_ONE
      - VALUE_TWO
      - VALUE_THREE
      type: string
  securitySchemes:
    SecurityScheme:
      type: http
      description: Authentication
      scheme: basic
```

### How to Reproduce?

Under https://github.com/KaiSuchomel/quarkus_openapi i have created a small reproducer.

1. Run Test `org.acme.CheckOpenApiTest`

With Quarkus 3.4.3 the CheckOpenApiTest is successfully running. With 3.5.0.CR1 not.

### Output of `uname -a` or `ver`

5.15.0-87-generic #97-Ubuntu SMP Mon Oct 2 21:09:21 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "17.0.6" 2023-01-17 OpenJDK Runtime Environment Temurin-17.0.6+10 (build 17.0.6+10) OpenJDK 64-Bit Server VM Temurin-17.0.6+10 (build 17.0.6+10, mixed mode, sharing)

### Quarkus version or git rev

3.5.0.CR1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.0 (9b58d2bad23a66be161c4664ef21ce219c2c8584)
Maven home: /opt/apache/maven/apache-maven-3.9.0
Java version: 17.0.6, vendor: Eclipse Adoptium, runtime: /home/kai/.jbang/cache/jdks/17
Default locale: de_DE, platform encoding: UTF-8
OS name: "linux", version: "5.15.0-87-generic", arch: "amd64", family: "unix"

**Comment by melloware** [comment]
@KaiSuchomel you might have to report that here if its an OpenAPI issue: https://github.com/smallrye/smallrye-open-api/issues

**Comment by KaiSuchomel** [comment]
@phillip-kruger Do you know in which quarkus version the fix will be available? I hope it will be 3.5.X.

**Comment by phillip-kruger** [comment]
I think we can include this in 3.5.1. @MikeEdgar ?

**Comment by MikeEdgar** [comment]
> I think we can include this in 3.5.1. @MikeEdgar ?

That's fine with me. I don't see any limitations around using SR OpenAPI 3.7 with any of the recent versions (aside from 3.2 since there are behavior changes).

**Comment by phillip-kruger** [comment]
Thanks @MikeEdgar for confirming. @KaiSuchomel I added the backport label on the PR that fix this, so depending on when we get that merged, it should be in a 3.5.x release
